core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -150,44 +150,127 @@ describe('CreateActivityButtonComponent', () => {
+      expect(component.canCreateCollections).toBe(false);
+
+      component.ngOnInit();
+      tick();
+      fixture.detectChanges();
+
+      expect(component.userIsLoggedIn).toBe(true);
+      expect(component.canCreateCollections).toBe(true);
+      expect(urlService.getUrlParams).toHaveBeenCalled();
+      expect(component.initCreationProcess).toHaveBeenCalled();
+    }));
+
+    it('should redirect user to creator dashboard page when the user' +
+      ' clicks on the create button', () => {
+      component.creationInProgress = false;
+      component.canCreateCollections = true;
+      spyOn(urlService, 'getPathname').and.returnValue('not/creator-dashboard');
+      spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+        location: {
+          replace: (val: string) => {}
+        }
+      });
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return <NgbModalRef>({
+          result: Promise.resolve('success')
+        });
+      });
+      const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
 
-    spyOn(component, 'initCreationProcess');
+      component.initCreationProcess();
 
-    expect(component.userIsLoggedIn).toBe(false);
-    expect(component.canCreateCollections).toBe(false);
+      expect(replaceSpy).toHaveBeenCalledWith('/creator-dashboard?mode=create');
+      expect(modalSpy).not.toHaveBeenCalled();
+    });
 
-    component.ngOnInit();
-    tick();
-    fixture.detectChanges();
+    it('should not redirect user but open a create activity modal if user' +
+      ' is on the creator dashboard page', () => {
+      component.creationInProgress = false;
+      component.canCreateCollections = true;
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/creator-dashboard');
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return <NgbModalRef>({
+          result: Promise.resolve('success')
+        });
+      });
+      const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
 
-    expect(component.userIsLoggedIn).toBe(true);
-    expect(component.canCreateCollections).toBe(true);
-    expect(component.initCreationProcess).toHaveBeenCalled();
-  }));
+      component.initCreationProcess();
 
-  it('should create a new exploration automatically if the user cannot create' +
-    ' collections and is redirected to creator dashboard after' +
-    ' logging in', fakeAsync(() => {
-    spyOn(userService, 'getUserInfoAsync')
-      .and.returnValue(Promise.resolve(userInfoForNonCollectionCreator));
-    const explorationCreationServiceSpy = spyOn(
-      explorationCreationService, 'createNewExploration');
+      expect(replaceSpy).not.toHaveBeenCalled();
+      expect(modalSpy).toHaveBeenCalled();
+    });
 
-    expect(component.userIsLoggedIn).toBe(false);
-    expect(component.canCreateCollections).toBe(false);
+    it('should handle modal\'s failure callback' +
```

Comment:
it should stop the creation process when there is an error with the XXX modal opening

(or similar -- "handle modal's failure callback is a bit too 'in the weeds'", and "handle" is not a specific word and should generally be avoided)

Reply:
Changed